components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emmits a new link submitted to a subreddit",
+  version: "0.0.2",
+  props: {
+    reddit,
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream poll for new links and emit accordingly with this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute
+      },
+    },
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "Enter subreddit name to listen for new links.",
+      default: "redditdev",
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emmits all existing events for the first time.
+      var after = null;
+      do {
```

Comment:
Good job on this one - we do like to emit sample events on the first run. I'd recommend only emitting 10 events on the first run, instead of all historical links. There's a timeout of 30 seconds on the deploy hook that users are likely to hit if you pull the whole history, and users typically won't need to see that many events - just a sample is appropriate.

Reply:
Thanks.

Reply:
So, at least in my local, I took care of only emitting 10 events on the deploy hook.
The next question derives from this change, would all the historical links be emitted in the first execution of the run event? is there a timeout  on run? reddit API allows a max limit of 100 for items to poll, so this could be added when making the request against.

Reply:
I also noticed that the first run event is executed automatically right after deploy. Is this exected? Is there a config on the source component to set a delay between the deploy hook and the first run?

Reply:
so, historical records, in general, are never emitted in sources? 
i can see an use case involving a Pipedream solution for the user could use a listing links action and have the source for new ones.  because of the changes I made historical records are not considered, just the sample of 10 records.

Reply:
so just confirming, we don't take any actions on historical records, other than the samples during deploy hook exec?

Reply:
as discussed on Slack, we don't need to worry about historical records.

```
what about the timestamp? in Reddit objects, it only has a "created" field which holds a timestamp. Should be put this in the ts property of the emmitting event? This is the original comment, bold for emphasis
yeah don't worry about historical records beyond the 10 sample records you're emitting now
```